Upgrade steps for Blazor WebAssembly (netstandard2.1 to .NET 5) are broken

Regarding the "Update Blazor WebAssembly projects" section, I attempted to follow those steps on a netstandard2.1 Blazor WebAssembly project that was generated a few months ago. The altered project failed to work, so next I tried creating a new ".NET Core 3.1" Blazor WebAssembly project from the standard template, and I repeated the steps again.

Both attempts failed, even though a .NET 5 Blazor Wasm project does work on my machine if it is created directly.

Based on my experience I think the following changes are needed:

1. Users should be instructed to Clean and Rebuild their project
2. Do not change the `<app>` element to `<div id="app">Loading...</div>` (in step 5) because the older project template is not designed to work with this change.
3. In Program.cs, do not change `"app"` to `"#app"` (in step 6)

Details follow.

### First Attempt (upgrade existing project) ###

After the first attempt, I got a long series of errors in the dev console of both Firefox and Chrome. It turned out that all of these errors were fixed after asking Visual Studio to "Clean" the project before building it. Here's how the errors looked in Firefox:

~~~
None of the “sha256” hashes in the integrity attribute match the content of the subresource. 2 localhost:52709
Uncaught (in promise) TypeError: NetworkError when attempting to fetch resource.
Uncaught (in promise) TypeError: NetworkError when attempting to fetch resource.
...
...
Error: Failed to start platform. Reason: TypeError: NetworkError when attempting to fetch resource.
    g http://localhost:52709/_framework/blazor.webassembly.js:1
    s http://localhost:52709/_framework/blazor.webassembly.js:1
    s http://localhost:52709/_framework/blazor.webassembly.js:1
    s http://localhost:52709/_framework/blazor.webassembly.js:1
blazor.webassembly.js:1:14681
~~~

And in Chrome:

~~~
Debugging hotkey: Shift+Alt+D (when application has focus)
localhost/:1 Failed to find a valid digest in the 'integrity' attribute for resource 'http://localhost:52709/_framework/Microsoft.AspNetCore.Authorization.dll' with computed SHA-256 integrity '9x0ibRAOMmEZK2MbZQ+HMnLABEaCh/R8ZHu6uDV6OFQ='. The resource has been blocked.
localhost/:1 Unknown error occurred while trying to verify integrity.
...
...
blazor.webassembly.js:1 Error: Failed to start platform. Reason: TypeError: Failed to fetch
    at blazor.webassembly.js:1
    at blazor.webassembly.js:1
    at Object.throw (blazor.webassembly.js:1)
    at s (blazor.webassembly.js:1)
~~~

After doing a Clean, these errors disappeared and a new problem occurred, which is the same problem I got in the Second Attempt:

### Second Attempt ###

This time I followed the upgrade steps directly on the .NET Core 3.1 Blazor WebAssembly project template. This time, the site's sidebar loads correctly (Home, Counter, Fetch Data tabs) but the page content is mangled - it looks blank, but it actually loads into a partially hidden region that is partially visible if you scroll down. No errors appear in the console:

~~~
blazor Loaded 8.63 MB resources
This application was built with linking (tree shaking) disabled. Published applications will be significantly smaller. blazor.webassembly.js:1:69909
mono_wasm_runtime_ready fe00e07a-5519-4dfe-b35a-f867dbaf2e28 dotnet.5.0.0.js:1:138552
~~~

I was able to fix the formatting by changing the main element back to `<app>Loading...</app>` and changing `"#app"` back to `"#app"` in Program.cs.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b031caa0-c62b-37da-46c3-59043256cac2
* Version Independent ID: bdf98c9a-96f6-79fd-a650-5c0385d1a62b
* Content: [Migrate from ASP.NET Core 3.1 to 5.0](https://docs.microsoft.com/en-us/aspnet/core/migration/31-to-50?view=aspnetcore-5.0&tabs=visual-studio#update-blazor-webassembly-projects)
* Content Source: [aspnetcore/migration/31-to-50.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/migration/31-to-50.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-migration**
* GitHub Login: @scottaddie
* Microsoft Alias: **scaddie**

--- a/aspnetcore/migration/31-to-50.md
+++ b/aspnetcore/migration/31-to-50.md
@@ -495,7 +495,7 @@ For more information, see:
 
 ### Clean and rebuild the solution
 
-After migrating the app to .NET 5, clean and rebuild the app. If package incompatibilities exist between new package references and cached packages:
+After migrating the app or solution to .NET 5, clean and rebuild the app or solution. If package incompatibilities exist between new package references and cached packages:
 
 1. Clear NuGet package caches by executing the following [`dotnet nuget locals`](/dotnet/core/tools/dotnet-nuget-locals) command in a command shell:
 

--- a/aspnetcore/migration/31-to-50.md
+++ b/aspnetcore/migration/31-to-50.md
@@ -325,6 +325,8 @@ If updating a Blazor WebAssembly project, skip to the [Update Blazor WebAssembly
 
 ## Update Blazor WebAssembly projects
 
+Follow the guidance in the preceding [Update Blazor WebAssembly and Blazor Server projects](#update-blazor-webassembly-and-blazor-server-projects) section.
+
 For a Blazor WebAssembly project, including the *`Client`* project of a hosted Blazor solution, apply the following changes to the project file:
 
 1. Update the SDK from `Microsoft.NET.Sdk.Web` to `Microsoft.NET.Sdk.BlazorWebAssembly`:
@@ -382,6 +384,8 @@ For a Blazor WebAssembly project, including the *`Client`* project of a hosted B
 
 ### Standalone Blazor WebAssembly app with Microsoft Accounts
 
+Follow the guidance in the preceding [Update Blazor WebAssembly and Blazor Server projects](#update-blazor-webassembly-and-blazor-server-projects) and [Update Blazor WebAssembly projects](#update-blazor-webassembly-projects) sections.
+
 For a standalone Blazor WebAssembly app registered in the Azure portal to use Azure Active Directory (AAD) for Microsoft Accounts:
 
 * The app requires the `openid` and `offline_access` scopes:
@@ -391,13 +395,17 @@ For a standalone Blazor WebAssembly app registered in the Azure portal to use Az
   options.ProviderOptions.DefaultAccessTokenScopes.Add("offline_access");
   ```
   
-* In the Azure portal app registration **Authentication** blade, set the platform configuration to **Single-page application** with the app's redirect URI.
+* In the Azure portal app registration **Authentication** blade:
 
-* Also in the **Authentication** blade, disable **Implicit grant** for **Access tokens** and **ID tokens**.
+  1. Remove the **Web** platform configuration.
+  1. Add a **Single-page application** platform configuration with the app's redirect URI.
+  1. Disable **Implicit grant** for **Access tokens** and **ID tokens**.
 
 For more information, see <xref:blazor/security/webassembly/standalone-with-microsoft-accounts>.
 
 ### Standalone Blazor WebAssembly app with Azure Active Directory (AAD)
+
+Follow the guidance in the preceding [Update Blazor WebAssembly and Blazor Server projects](#update-blazor-webassembly-and-blazor-server-projects) and [Update Blazor WebAssembly projects](#update-blazor-webassembly-projects) sections.
 
 For a standalone Blazor WebAssembly app registered in the Azure portal to use Azure Active Directory (AAD):
 
@@ -408,13 +416,17 @@ For a standalone Blazor WebAssembly app registered in the Azure portal to use Az
       .Add("https://graph.microsoft.com/User.Read");
   ```
   
-* In the Azure portal app registration **Authentication** blade, set the platform configuration to **Single-page application** with the app's redirect URI.
+* In the Azure portal app registration **Authentication** blade:
 
-* Also in the **Authentication** blade, disable **Implicit grant** for **Access tokens** and **ID tokens**.
+  1. Remove the **Web** platform configuration.
+  1. Add a **Single-page application** platform configuration with the app's redirect URI.
+  1. Disable **Implicit grant** for **Access tokens** and **ID tokens**.
   
 For more information, see <xref:blazor/security/webassembly/standalone-with-azure-active-directory>.
 
-### Standalone Blazor app with Azure Active Directory (AAD) B2C
+### Standalone Blazor WebAssembly app with Azure Active Directory (AAD) B2C
+
+Follow the guidance in the preceding [Update Blazor WebAssembly and Blazor Server projects](#update-blazor-webassembly-and-blazor-server-projects) and [Update Blazor WebAssembly projects](#update-blazor-webassembly-projects) sections.
 
 For a standalone Blazor WebAssembly app registered in the Azure portal to use Azure Active Directory (AAD) B2C:
 
@@ -425,26 +437,24 @@ For a standalone Blazor WebAssembly app registered in the Azure portal to use Az
   options.ProviderOptions.DefaultAccessTokenScopes.Add("offline_access");
   ```
   
-* In the Azure portal app registration **Authentication** blade, set the platform configuration to **Single-page application** with the app's redirect URI.
+* In the Azure portal app registration **Authentication** blade:
 
-* Also in the **Authentication** blade, disable **Implicit grant** for **Access tokens** and **ID tokens**.
+  1. Remove the **Web** platform configuration.
+  1. Add a **Single-page application** platform configuration with the app's redirect URI.
+  1. Disable **Implicit grant** for **Access tokens** and **ID tokens**.
 
 For more information, see <xref:blazor/security/webassembly/standalone-with-azure-active-directory-b2c>.
 
-### Hosted Blazor WebAssembly app with Azure Active Directory (AAD) or B2C
-
-The *`Client`* app registration of a hosted Blazor solution that uses AAD or AAD B2C for user authentication should use a **Single-page application** Azure Apps platform configuration:
-
-1. In the Azure portal app registration for the *`Client`* app, remove the **Web** platform configuration.
-1. Add a **Single-page application** platform configuration with the app's redirect URI.
-1. Disable **Implicit grant** for **Access tokens** and **ID tokens**.
-
-For more information, see:
-
-* <xref:blazor/security/webassembly/hosted-with-azure-active-directory>
-* <xref:blazor/security/webassembly/hosted-with-azure-active-directory-b2c>
-
 ### Update the Server project of a hosted Blazor solution
+
+Follow the guidance in the preceding sections:
+
+* [Update Blazor WebAssembly and Blazor Server projects](#update-blazor-webassembly-and-blazor-server-projects)
+* [Update Blazor WebAssembly projects](#update-blazor-webassembly-projects) sections
+* The section that applies to the app's provider with Azure Active Directory:
+  * [Standalone Blazor WebAssembly app with Microsoft Accounts](#standalone-blazor-webassembly-app-with-microsoft-accounts)
+  * [Standalone Blazor WebAssembly app with Azure Active Directory (AAD)](#standalone-blazor-webassembly-app-with-azure-active-directory-aad)
+  * [Standalone Blazor WebAssembly app with Azure Active Directory (AAD) B2C](#standalone-blazor-webassembly-app-with-azure-active-directory-aad-b2c)
 
 Update the *`Server`* project of a hosted Blazor solution as an ASP.NET Core app following the general guidance in this article.
 
@@ -482,6 +492,35 @@ For more information, see:
 
 * <xref:blazor/security/webassembly/hosted-with-azure-active-directory>
 * <xref:blazor/security/webassembly/hosted-with-azure-active-directory-b2c>
+
+### Clean and rebuild the solution
+
+After migrating the app to .NET 5, clean and rebuild the app. If package incompatibilities exist between new package references and cached packages:
+
+1. Clear NuGet package caches by executing the following [`dotnet nuget locals`](/dotnet/core/tools/dotnet-nuget-locals) command in a command shell:
+
+   ```dotnetcli
+   dotnet nuget locals --clear all
+   ```
+   
+1. Clean and rebuild the app or solution.
+
+### Troubleshoot
+
+Follow the *Troubleshoot* guidance at the end of the Blazor WebAssembly security topic that applies to your app:
+
+Standalone Blazor WebAssembly apps:
+
+* [General guidance for OIDC providers and the WebAssembly Authentication Library](xref:blazor/security/webassembly/standalone-with-authentication-library)
+* [Microsoft Accounts](xref:blazor/security/webassembly/standalone-with-microsoft-accounts)
+* [Azure Active Directory (AAD)](xref:blazor/security/webassembly/standalone-with-azure-active-directory)
+* [Azure Active Directory (AAD) B2C](xref:blazor/security/webassembly/standalone-with-azure-active-directory-b2c)
+
+Hosted Blazor WebAssembly apps:
+
+* [Azure Active Directory (AAD)](xref:blazor/security/webassembly/hosted-with-azure-active-directory)
+* [Azure Active Directory (AAD) B2C](xref:blazor/security/webassembly/hosted-with-azure-active-directory-b2c)
+* [Identity Server](xref:blazor/security/webassembly/hosted-with-identity-server)
 
 ### Unauthorized client for Azure Active Directory (AAD)
 

